src/V3LinkJump.cpp

```diff
@@ -83,6 +83,14 @@ class LinkJumpVisitor final : public VNVisitor {
                 underp = nodep;
                 under_and_next = false;  // IE we skip the entire while
             }
+        } else if (VN_IS(nodep, DoWhile)) {
+            // Handle it the same as AstWhile, because it will be converted to it
+            if (endOfIter) {
+                underp = VN_AS(nodep, DoWhile)->stmtsp();
```

Comment:
```suggestion
        } else if (AstDoWhile* const dowhilep = VN_CAST(nodep, DoWhile)) {
            // Handle it the same as AstWhile, because it will be converted to it
            if (endOfIter) {
                underp = dowhilep->stmtsp();
```